cups/sidechannel.c

```diff
@@ -226,7 +226,7 @@ cupsSideChannelRead(
   * Validate the data length in the message...
   */
 
-  templen = ((buffer[2] & 255) << 8) | (buffer[3] & 255);
+  templen = (((unsigned char)buffer[2]) << 8) | ((unsigned char)buffer[3]);
```

Comment:
not if casted to unsigned char first

Reply:
Nope, if you cast the pointer and then dereference, you're good. Otherwise you just get the sign-extended (if char is signed) version with a value > 255.

Reply:
Alright, thanks!